main/js/afterSubmit.js

```diff
@@ -187,16 +187,28 @@ const buildHeatmap = function(expData, clinData) {
  */
 const buildViolinPlot = function(geneQuery, expressionData) {
   //Remove loader from violin plot container
-  document.getElementById("violinLoaderDiv").classList.remove("loader");
+  var violinLoaderDiv = document.getElementById("violinLoaderDiv");
+  violinLoaderDiv.classList.remove("loader");
 
-  // Create partition selector
-  const partitionDivId = "violinPartition";
-  addDivInside(partitionDivId, "violinLoaderDiv");
-  // Create partition selector
-  createViolinPartitionBox("violinPlots", geneQuery);
+  //Setup Materialize Grid
+  addDivInside("violinGridRow", violinLoaderDiv.id);
+  var gridRow = document.getElementById("violinGridRow");
+  gridRow.classList.add("row");
 
-  // Create div for violin plots
-  addDivInside("violinPlots", "violinLoaderDiv");
+  // Create partition selector div and add it inside Materialize Grid
+  const partitionDivId = "violinPartition";
+  addDivInside(partitionDivId, gridRow.id);
+  var partitionCol = document.getElementById(partitionDivId);
+  partitionCol.classList.add("col");
+  partitionCol.classList.add("s3");
```

Comment:
minor thing but we can add multiple classes with one call to `.add()`

```suggestion
  partitionCol.classList.add("col", "s3");
```